bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/BsonEncoder.kt

```diff
@@ -37,31 +38,57 @@ import org.bson.types.ObjectId
  *
  * For custom serialization handlers
  */
-public sealed interface BsonEncoder {
+@ExperimentalSerializationApi
+internal sealed interface BsonEncoder : Encoder, CompositeEncoder {
```

Comment:
Same as the BsonDecoder interface - its sealed and had no need to be public.

Reply:
I'll revert as its not strictly true - users could in theory use this aside from the Codecs.